src/app/plugins/weather.plugin.ts

```diff
@@ -0,0 +1,270 @@
+
+    const embed: RichEmbed = new RichEmbed();
+
+    const wdata = JSON.parse(JSON.stringify(wrawdata));
+    const fdata = JSON.parse(JSON.stringify(frawdata));
+    const country: string = wdata.sys.country.toLowerCase();
+    const weather_code: number = wdata.weather[0].id;
+    const title_desc: string = `${this.getTempComment(weather_code,wdata.main.temp)}\n${this.getWindComment(wdata.wind.speed)}`;    
+    const desc: string = this.capitalize(wdata.weather[0].description);
+
+    let s: string = "";
+    const date: Date = new Date();
+    const diff: number = Math.ceil(((date.getTime()/1000) - Number(wdata.dt))/60);
+    if(diff>1)
+        s = "s";
+    embed.setFooter(`Last Updated ${diff} minute${s} ago`);
+    embed.setTimestamp(wdata['dt']*1000);
+    embed.addField(`${wdata.name} :flag_${country}:`,`${title_desc}`);
+    embed.setColor("#ffee05");
+    embed.setThumbnail(`https://www.gstatic.cn/onebox/weather/128/${this.getWeatherIcon(weather_code)}.png`);
+    embed.addField("Temperature",`🌡 ${Math.floor(wdata.main.temp)}°F (${Math.floor(((wdata.main.temp-32)*5)/9)}°C)`, true);
+    embed.addField("Description",`📢 ${desc}`, true);
+
+    let rain: number = 0;
+    let rain_resp: string = "last hour";
+    if(wdata.rain != null){
+        if(wdata.rain["1h"] != null){
+            rain = wdata['rain']['1h'];
+        }else{
+            rain = wdata['rain']['3h'];
+            rain_resp = "last 3 hours";
+        }
+    }else{
+        rain = 0;
+    }
+    embed.addField("Rainfall",`${rain}mm in ${rain_resp}`,true);
+
+    let wind_dir : string = "";
+    if(wdata.wind != null){
+        if(wdata.wind.deg != null)
+        wind_dir = this.getWindArrow(wdata.wind['deg']);
+    }     
+    else 
+        wind_dir = this.getWindArrow(-1);
+    embed.addField("Wind",`💨 ${Math.floor(wdata.wind.speed)} ${wind_dir}`,true);
+
+    embed.addField("​","​",false);
+    embed.addField("Forecast",this.generateForecast(fdata));
+    return embed;
+  }
+  
+  private generateForecast(fdata:any):string{
+    let output: string = "";
+    let x: number = 0;
+    let xc: number = 0;
+
+    while(xc < this.forecast_num){
+        let tdata = fdata['list'][x];
+        let timestamp = tdata['dt'];
+        let date = new Date();
+        let hours: number = Math.ceil((timestamp -  Math.round(Date.now()/1000))/3600); 
+
+        if(hours > 0){
+            let dt = new Date(timestamp*1000);
+            let strf:string = dt.getHours().toString();
+            strf += (Math.floor(dt.getHours()/12) < 1 ? "AM":"PM");
+
+            let weather_id = tdata.weather[0].id;
+
+            output += `**${hours} hours (${strf})** - `;
+            output += `${Math.round(tdata.main.temp)} °F`;
+            output += `   -   ${this.getWeatherEmoji(weather_id)} ${this.capitalize(tdata.weather[0].description)}\n`
+            xc += 1;
+        }
+        x += 1
+       
+    }
+    
+    return output
+  }
+  private getWeatherEmoji(code:number):string{
```

Comment:
Clean up all the `if`/`else` stuff by using some arrays, loops, maps or something.